agent/acs/handler/payload_handler.go

```diff
@@ -274,7 +286,7 @@ func (payloadHandler *payloadRequestHandler) handleUnrecognizedTask(task *ecsacs
 	}
 
 	// Only need to stop the task; it brings down the containers too.
-	eventhandler.AddTaskEvent(api.TaskStateChange{
+	payloadHandler.taskHandler.AddTaskEvent(api.TaskStateChange{
```

Comment:
Is this the only reason that `taskHandler` needs to be part of the `payloadHandler`? We may implement this by using the `taskEngine.TaskEvents()`, since `taskEngine` is already part of the `payloadHandler` so that `payloadHandler` can get rid of the `taskHandler`?